Automattic/jetpack
# Ensure correct reporting for Cart/Checkout shortcode/blocks properties in WooCommerce analytics events

## Proposed changes:

* If WooCommerce Blocks is version 10.6.0 or above, it will be using the Cart and Checkout templates to power those WooCommerce pages. In this case, the incorrect values may be reported for the following properties on `woocommerceanalytics_%` events:
  * `cart_page_contains_cart_block`
  * `cart_page_contains_cart_shortcode`
  * `checkout_page_contains_cart_block`
  * `checkout_page_contains_cart_shortcode`
* The changes introduced in this PR are:
* Get the version of WC blocks, if it is greater than 10.6 load the templates from the WC Blocks `BlockTemplatesController` - load the Cart and Checkout templates and check what they contain. This is the same logic as the existing checks, however the string (page content vs. template content) being searched is different.


### Other information:


## Jetpack product discussion



A previous discussion about this tracking can be found here pb0Spc-ou-p2#comment-701 however it is very old.

We also have pca54o-6dR-p2 where the effort is listed as a cycle goal (point 3).

## Does this pull request change what data or activity we track or use?


No, it just ensures the data being tracked is accurate, but the data is not changed.

## Testing instructions:







- Ensure site is connected, WooCommerce is installed, products, payment methods, and shipping methods are available. (Cash on Delivery and Free shipping will be fine).
- Install WooCommerce 7.8.0 and a blocks theme e.g. Twenty Twenty-Three.
- Ensure WooCommerce analytics is running.
- Go to Pages -> Cart. Change its content to contain the shortcode block, and have it display the WooCommerce cart. (`[woocommerce_cart]`).
- Do the same for Pages -> Checkout and enter `[woocommerce_checkout]` into the shortcode block.
- Add an item to your cart and go to the checkout.
- Check out and then visit Tracks and find your event. (I spoofed my user agent so I could find the event easily)
- Check the event for the following properties:
  - `cart_page_contains_cart_block`: `false` (or 0)
  - `cart_page_contains_cart_shortcode`: `true` (or 1)
  - `checkout_page_contains_checkout_block`: `false` (or 0)
  - `checkout_page_contains_checkout_shortcode`: true (or 1)
- Go back to the pages from the earlier steps, remove the shortcodes and replace them with the Cart and Checkout blocks.
- Add an item to your cart and check out again, the new values should be:
  - `cart_page_contains_cart_block`: true (or 1)
  - `cart_page_contains_cart_shortcode`: false (or 0)
  - `checkout_page_contains_checkout_block`: true (or 1)
  - `checkout_page_contains_checkout_shortcode`: false (or 0)
- Feel free to change the setup so one page has the shortcode and one page doesn't and mix it up.
- Update to WooCommerce 8.0
- Repeat the steps above, however when you go to Pages -> Checkout and Pages -> Cart you should notice that it opens the site editor instead of the usual post editor.
- Repeat with a classic theme, e.g Storefront
